tests/network/primary_pod_network.go

```diff
@@ -70,25 +70,69 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 		})
 
 		Context("VMI connected to the pod network using bridge binding", func() {
-			var vmi *v1.VirtualMachineInstance
+			When("Guest Agent exists", func() {
+				var (
+					vmi   *v1.VirtualMachineInstance
+					vmiIP = func() (string, error) {
+						vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+						if err != nil {
+							return "", err
+						}
+						return vmi.Status.Interfaces[0].IP, nil
+					}
 
-			BeforeEach(func() {
-				vmi = setupVMI(virtClient, vmiWithBridgeBinding())
-			})
+					vmiIPs = func() ([]string, error) {
+						vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+						if err != nil {
+							return nil, err
+						}
+						return vmi.Status.Interfaces[0].IPs, nil
+					}
+				)
+				BeforeEach(func() {
+					var err error
 
-			AfterEach(func() {
-				cleanupVMI(virtClient, vmi)
+					vmi, err = newFedoraWithGuestAgentAndDefaultInterface(libvmi.InterfaceDeviceWithBridgeBinding())
+					Expect(err).NotTo(HaveOccurred())
+
+					vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+					Expect(err).NotTo(HaveOccurred())
+					tests.WaitForSuccessfulVMIStart(vmi)
+					tests.WaitAgentConnected(virtClient, vmi)
+				})
+
+				AfterEach(func() {
+					cleanupVMI(virtClient, vmi)
+				})
+
+				It("should report PodIP/s as its own on interface status", func() {
+					vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
+					Eventually(vmiIP).Should(WithTransform(removeCIDR, Equal(vmiPod.Status.PodIP)), "should contain VMI Status IP as Pod status ip")
```

Comment:
correct me please if i am wrong,
its DefaultPodNetwork, so how its connected to guest agent ?
aint it just as `ValidateVMIandPodIPMatch` ?

i would add a 2nd interface check that networkData update it and guest agent reports it, unless thats what you do already good enough.

P.S
since you use removeCIDR, i do understand guest agent is involved because its the one that adds the subnet,
on the other hand please see the comment about removeCIDR and i would suggest to make sure we check here guest agent reported data please.

Reply:
@oshoval I do not understand what you are saying here. Please rephrase.
This assert checks that the vmi IP, without the prefix part, is equal to the Pod IP.

What part you see problematic here?

Reply:
I think i am wrong (see please first line)
i am used that we dont change the default interface with guest agent
but from what Quique explained we do have here a bridge binding that is set by cloud-init IIUC